### PR TITLE
[RestorePoints] Adding Encryption, Source Details, HyperVGeneration and WriteAccelerated for Restore Point

### PR DESCRIPTION
**Pull request by linugeorgeofficial**
## ARM API Information (Control Plane)

<i>MSFT employees can try out our new experience at <b>[OpenAPI Hub](https://aka.ms/openapiportal) </b> - one location for using our validation tools and finding your workflow.
</i>
Azure 1st Party Service can try out the [Shift Left](https://aka.ms/ShiftLeft) experience to initiate API design review from ADO code repo.  If you are interested, may request engineering support by filling in with the form https://aka.ms/ShiftLeftSupportForm.

### Changelog
Add a changelog entry for this PR by answering the following questions:
  1. What's the purpose of the update?
      - [ ] new service onboarding
      - [ ] new API version
      - [ ] update existing version for new feature
      - [ ] update existing version to fix swagger quality issue in s360
      - [ ] Other, please clarify
  2. When are you targeting to deploy the new service/feature to public regions? Please provide the date or, if the date is not yet available, the month.
  3. When do you expect to publish the swagger? Please provide date or, the the date is not yet available, the month.
  4. By default, Azure SDKs of all languages (.NET/Python/Java/JavaScript for both management-plane SDK and data-plane SDK, Go for management-plane SDK only ) MUST be refreshed with/after swagger of new version is published. If you prefer NOT to refresh any specific SDK language upon swagger updates in the current PR, please leave details with justification here.

### Contribution checklist (MS Employees Only):
- [ ] I commit to follow the [Breaking Change Policy](https://aka.ms/AzBreakingChangesPolicy) of "no breaking changes"
- [ ] I have reviewed the [documentation](https://aka.ms/ameonboard) for the workflow.
- [ ] [Validation tools](https://aka.ms/swaggertools) were run on swagger spec(s) and errors have all been fixed in this PR. [How to fix?](https://aka.ms/ci-fix)

If any further question about AME onboarding or validation tools, please view the [FAQ](https://aka.ms/faqinprreview).

### ARM API Review Checklist

> **Applicability**: :warning:
>
> If your changes encompass only the following scenarios, you should SKIP this section, as these scenarios do not require ARM review.
> - Change to data plane APIs
> - Adding new properties
> - All removals

Otherwise your PR may be subject to ARM review requirements. Complete the following:
- [ ] Check this box if any of the following apply to the PR so that the label "ARMReview" and "WaitForARMFeedback" will be added by bot to kick off ARM API Review.  Missing to check this box in the following scenario may result in delays to the ARM manifest review and deployment.
  - Adding a new service
  - Adding new API(s)
  - Adding a new API version
    -[ ] To review changes efficiently, ensure you copy the existing version into the new directory structure for first commit and then push new changes, including version updates, in separate commits. You can use OpenAPIHub to initialize the PR for adding a new version. For more details refer to the [wiki](https://dev.azure.com/azure-sdk/internal/_wiki/wikis/internal.wiki/208/OpenAPI-Hub-Adding-new-API-version).

- [ ] Ensure you've reviewed following [guidelines](https://aka.ms/rpguidelines) including [ARM resource provider contract](https://github.com/Azure/azure-resource-manager-rpc) and [REST guidelines](https://github.com/microsoft/api-guidelines/blob/vNext/azure/Guidelines.md). Estimated time (4 hours). This is required before you can request review from ARM API Review board.

- [ ] If you are blocked on ARM review and want to get the PR merged with urgency, please get the ARM oncall for reviews (*RP Manifest Approvers* team under <ins>Azure Resource Manager service</ins>) from IcM and reach out to them.

### Breaking Change Review Checklist
If you have any breaking changes as defined in the [Breaking Change Policy](https://aka.ms/AzBreakingChangesPolicy/), request approval from the Breaking Change Review Board.

**Action**: to initiate an evaluation of the breaking change, create a new intake using the [template for breaking changes](https://aka.ms/Breakingchangetemplate). Additional details on the process and office hours are on the [Breaking Change Wiki](https://dev.azure.com/msazure/AzureWiki/_wiki/wikis/AzureWiki.wiki/37684/Breaking-Changes).

NOTE: To update API(s) in public preview for over 1 year (refer to [Retirement of Previews](https://dev.azure.com/msazure/AzureWiki/_wiki/wikis/AzureWiki.wiki/37683/Retirement-of-Previews))

Please follow the link to find more details on [PR review process](https://aka.ms/SwaggerPRReview).

### CODE REVIEW

File: specification/compute/resource-manager/Microsoft.Compute/ComputeRP/stable/2023-03-01/restorePoint.json [hunk 1]
```diff
@@ -775,12 +775,67 @@
           "description": "Gets the managed disk details"
         },
         "diskRestorePoint": {
-          "$ref": "./computeRPCommon.json#/definitions/ApiEntityReference",
+          "$ref": "#/definitions/DiskRestorePointProperties",
           "description": "Gets the disk restore point Id."
```
Comment (iamashwani-who) [inline specification/compute/resource-manager/Microsoft.Compute/ComputeRP/stable/2023-03-01/restorePoint.json:779]: We can change the description as it is not just disk restore point Id

File: specification/compute/resource-manager/Microsoft.Compute/ComputeRP/stable/2023-03-01/restorePoint.json [hunk 2]
```diff
@@ -775,12 +775,67 @@
...
         }
       },
       "description": "Describes a data disk."
     },
+    "DiskRestorePointProperties": {
+      "properties": {
+        "encryption": {
+          "$ref": "#/definitions/Encryption"
```
Comment (iamashwani-who) [inline specification/compute/resource-manager/Microsoft.Compute/ComputeRP/stable/2023-03-01/restorePoint.json:787]: can we add description that it is optional

**Review by iamashwani-who — APPROVED** [review]

File: specification/compute/resource-manager/Microsoft.Compute/ComputeRP/stable/2023-03-01/restorePoint.json (hunk 1 above)
Comment (linugeorgeofficial) [reply specification/compute/resource-manager/Microsoft.Compute/ComputeRP/stable/2023-03-01/restorePoint.json]: Changed

**Comment by grizzlytheodore** [comment]
@JeffreyRichter please approve the breaking changes.

here are the justifications provided by the service team, and also **the feature is in preview:** 
**Justification for the DiskRestorePointAttributes change:** 
Previously, once customer creates a restore point, the only property specific to the disk present was the id (ApiEntityReference). 
However, there is an additional properties (encryption) of the disk restore point that should be added to the restore point, as the disk restore point can have different encryption that the source disk. The only appropriate place to add it without redundancy is to add it along with diskRestorePoint.id . For this, Instead of diskRestorePoint being an ApiEntityReference, it needs to be another object that inherits ApiEntityReference. 
**Justification for the readOnly change:** 
Until now there was no way for the customer to specify what encryption settings they wanted on their disk restore points. With this change ([Commit fec0dd39: Merged PR 7471733: [Restore Points] Support for Encryption Input for Restore... - Repos (visualstudio.com)](https://nam06.safelinks.protection.outlook.com/?url=https%3A%2F%2Fmsazure.visualstudio.com%2FOne%2F_git%2FCompute-CPlat-Core%2Fcommit%2Ffec0dd3979753896ad202a2ee6aee252d59f45a3%3FrefName%3Drefs%2Fheads%2Fmaster&data=05%7C01%7CTheodore.Chang%40microsoft.com%7Ca34cc55e12d14e65641e08db35258805%7C72f988bf86f141af91ab2d7cd011db47%7C1%7C0%7C638162207604383034%7CUnknown%7CTWFpbGZsb3d8eyJWIjoiMC4wLjAwMDAiLCJQIjoiV2luMzIiLCJBTiI6Ik1haWwiLCJXVCI6Mn0%3D%7C3000%7C%7C%7C&sdata=yQX2QYRemlAHB4xvCDsqj9XLh93Jwep9vrKqMpKsaKo%3D&reserved=0)), we now let customer specify encryption settings for local and cross region restore points. To do this, they need to tell us the restore point of which source disk they want to apply these encryption settings, along with the actual encryption settings that they want. So, ManagedDisk.id or DiskRestorePoint.sourceDiskRestorePoint is needed as input to identify source (local and cross region respectively), and DiskRestorePoint.encryption is needed for the actual encryption settings.

**Comment by JeffreyRichter** [comment]
Can you come to our Office Hours on Monday @ 10am Pacific to walk us through these breaks?
Changing so many properties to readonly is severely breaking. And when I look at the list of breaks, I don't even see the DiskRestorePointAttributes change you're referring to.

**Comment by grizzlytheodore** [comment]
> Can you come to our Office Hours on Monday @ 10am Pacific to walk us through these breaks?
> 
> Changing so many properties to readonly is severely breaking. And when I look at the list of breaks, I don't even see the DiskRestorePointAttributes change you're referring to.
> 
> 

@JeffreyRichter could you please share meeting invite? thchan@microsoft.com
linugeorge@microsoft.com

**Comment by grizzlytheodore** [comment]
@JeffreyRichter  also, to help you understand whats going on for the readOnly tags: 
a.	[RestorePointSourceMetadata](https://nam06.safelinks.protection.outlook.com/?url=https%3A%2F%2Fgithub.com%2FAzure%2Fazure-rest-api-specs%2Fpull%2F23303%2Ffiles%23diff-0bbdea08077b1e850e5da5cc5571b930c605f615fec80f61b5a47a920cc68498R651&data=05%7C01%7CTheodore.Chang%40microsoft.com%7Ca34cc55e12d14e65641e08db35258805%7C72f988bf86f141af91ab2d7cd011db47%7C1%7C0%7C638162207604383034%7CUnknown%7CTWFpbGZsb3d8eyJWIjoiMC4wLjAwMDAiLCJQIjoiV2luMzIiLCJBTiI6Ik1haWwiLCJXVCI6Mn0%3D%7C3000%7C%7C%7C&sdata=X%2BG22qKwYDFPY9xAOEyL8Nas5E0CW5fTFVS%2BA28j9D0%3D&reserved=0) was entirely readOnly up until now. That means, everything under it was readOnly.
With this PR ([Commit fec0dd39: Merged PR 7471733: [Restore Points] Support for Encryption Input for Restore... - Repos (visualstudio.com)](https://nam06.safelinks.protection.outlook.com/?url=https%3A%2F%2Fmsazure.visualstudio.com%2FOne%2F_git%2FCompute-CPlat-Core%2Fcommit%2Ffec0dd3979753896ad202a2ee6aee252d59f45a3%3FrefName%3Drefs%2Fheads%2Fmaster&data=05%7C01%7CTheodore.Chang%40microsoft.com%7Ca34cc55e12d14e65641e08db35258805%7C72f988bf86f141af91ab2d7cd011db47%7C1%7C0%7C638162207604383034%7CUnknown%7CTWFpbGZsb3d8eyJWIjoiMC4wLjAwMDAiLCJQIjoiV2luMzIiLCJBTiI6Ik1haWwiLCJXVCI6Mn0%3D%7C3000%7C%7C%7C&sdata=yQX2QYRemlAHB4xvCDsqj9XLh93Jwep9vrKqMpKsaKo%3D&reserved=0)), we now allow customer to give “[DiskRestorePointProperties](https://nam06.safelinks.protection.outlook.com/?url=https%3A%2F%2Fgithub.com%2FAzure%2Fazure-rest-api-specs%2Fpull%2F23303%2Ffiles%23diff-0bbdea08077b1e850e5da5cc5571b930c605f615fec80f61b5a47a920cc68498R801&data=05%7C01%7CTheodore.Chang%40microsoft.com%7Ca34cc55e12d14e65641e08db35258805%7C72f988bf86f141af91ab2d7cd011db47%7C1%7C0%7C638162207604383034%7CUnknown%7CTWFpbGZsb3d8eyJWIjoiMC4wLjAwMDAiLCJQIjoiV2luMzIiLCJBTiI6Ik1haWwiLCJXVCI6Mn0%3D%7C3000%7C%7C%7C&sdata=iq8q2iM97OPAV9aQEM95xmB%2B3I0ZOVUczoFudxwJY1M%3D&reserved=0)” and “[managedDisk](https://nam06.safelinks.protection.outlook.com/?url=https%3A%2F%2Fgithub.com%2FAzure%2Fazure-rest-api-specs%2Fpull%2F23303%2Ffiles%23%3A~%3Atext%3D%257D%252C-%2C%2522managedDisk%2522%2C-%253A%2520%257B&data=05%7C01%7CTheodore.Chang%40microsoft.com%7Ca34cc55e12d14e65641e08db35258805%7C72f988bf86f141af91ab2d7cd011db47%7C1%7C0%7C638162207604383034%7CUnknown%7CTWFpbGZsb3d8eyJWIjoiMC4wLjAwMDAiLCJQIjoiV2luMzIiLCJBTiI6Ik1haWwiLCJXVCI6Mn0%3D%7C3000%7C%7C%7C&sdata=mmmG8DPXRpUZiRy4OEPcFfBqle67UQ2vPhsrYSDElWc%3D&reserved=0)” as input. Thus, the readOnly tag was removed from RestorePointSourceMetadata, and added to every other property under it explicitly, so that only these would be the new enterable fields.

**Review by prchin — APPROVED** [review]

**Pull request merged**